forte/data/readers/wiki_passage_qa_reader.py

```diff
@@ -0,0 +1,159 @@
+                          just one line of text)
+                          (4) documents associated with a query in case the
+                          data_pack is of type Query
+        """
+        dir_path: str = args[0]
+        split: str = args[1]
+
+        corpus_file_path = os.path.join(dir_path, 'document_passages.json')
+        split_file_path = os.path.join(dir_path, f'{split}.tsv')
+
+        passage_mode = self.configs.doc_mode == "passage"
+
+        with open(corpus_file_path, 'r') as f:
+            corpus = json.loads(f.read())
+
+        df = pd.read_csv(split_file_path, sep='\t')
+
+        def _get_passage_ids(row_):
+            did = f'{row_["DocumentID"]}'
+            return [f'{did.strip()}_{pid.strip()}'
+                    for pid in f'{row_["RelevantPassages"]}'.split(',')]
+
+        if passage_mode:
+            df["labels"] = df.apply(_get_passage_ids, axis=1)
+        else:
+            df["labels"] = df["DocumentID"].apply(lambda lst: [f'{lst}'])
+
+        # send all the queries with relevant docs
+        for _, row in df.iterrows():
+            yield True, row["QID"], [row["Question"]], row["labels"]
+
+        # send all the docs
+        for _, row in df.iterrows():
+            for doc_id in row["labels"]:
+                if passage_mode:
+                    did, pid = doc_id.split('_')
+                    doc_content = [corpus[did][pid]]
+                else:
+                    doc = corpus[doc_id]
+                    doc_content = [doc[pid] for pid in sorted(doc.keys())]
+                yield False, doc_id, doc_content, None
+
+    def _parse_pack(self, doc_info: DocInfoType) -> Iterator[DataPack]:
+        # pylint: disable = no-self-use
+        """
+        Takes the `doc_info` returned by the `_collect` method and returns a
+        `data_pack` that either contains entry of the type `Query`, or contains
+        an entry of the type Document.
+        :param doc_info: document info to be populatd in the data_pack
+        :return: query or document data_pack
+        """
+        data_pack: DataPack = DataPack()
+
+        is_query, doc_id, doc_content, rel_docs = doc_info
+        data_pack.meta.doc_id = doc_id
+
+        annotations: List[Union[Passage, Document, Query]] = []
+
+        if not is_query:
+            # add passages
+            for passage in doc_content:
+                annotations.append(Passage(data_pack, 0, len(passage)))
+
+            doc_text = os.linesep.join(doc_content)
+            annotations.append(Document(data_pack, 0, len(doc_text)))
+            data_pack.set_text(doc_text)
+        else:
+            query = Query(data_pack)
+            query.query = doc_content[0]
+            if rel_docs is not None:
+                query.doc_ids = {"relevant_docs": rel_docs}
+            annotations = [query]
+
+        # add annotations to data_pack
+        for annotation in annotations:
+            data_pack.add_or_get_entry(annotation)
```

Comment:
Let's don't use ``add_or_get_entry`` in the future, we are planning to deprecate them.